base/task.jl

```diff
@@ -1176,3 +1206,20 @@ if Sys.iswindows()
 else
     pause() = ccall(:pause, Cvoid, ())
 end
+
+function record_cpu_time!(t::Task, stopped_at::UInt64=time_ns())
+    if t.is_timing_enabled
+        @assert t.last_dequeued_at != 0
+        t.cpu_time_ns += stopped_at - t.last_dequeued_at
+        t.last_dequeued_at = 0
```

Comment:
Do you ever check/rely anywhere that this is reset to 0?

It seems unnecessary to me (and we want to keep task-switches as cheap as we can, even though this will probably be very cheap since we'll already have the task resident in-cache, but still writes to main memory are slowish). And maybe someone would even want to look at the last-ran-time even for a sleeping task if they were trying to debug things? Not sure though.

Reply:
we now use it in `Base.task_cpu_time_ns` to determine if the task is running, so that can report the CPU time of running tasks (without it being out of date til the task yields or finishes)

We could have split added a `task_state_running` for this (to distinguish running from `task_state_runnable`) but that felt like a more intrusive change. We could have added a `is_running::Bool` field for it, but it seemed simpler to me to re-use this field.

Reply:
update: ignore what i said. removed these after all